added parabolic region approach

nikita-astronaut opened the pull request:
Hello! I added parabolic region approach and tests to it. Seems like everything works.

However, in hyperbolic region, if we take their initial guess, it fails for high ecc.

So, it indeed works faster for small eccs, but for ecc like 3200, Newton can not converge with their initial guess. SO I decided not to add it, at least for now.

Juanlu001 commented on src/poliastro/twobody/angles.py:
```diff
@@ -24,6 +24,126 @@ def _kepler_equation_prime_hyper(F, M, ecc):
+
+    Note
+    -----
+    Taken from the paper <<Robust resolution of Kepler’s equation in all eccentricity regimes>>,
```
Could you change the citation style here to something more standard? For example (taken from scholar.google.com, Chicago style):

```
.. [1] Farnocchia, Davide, Davide Bracali Cioci, and Andrea Milani.
   "Robust resolution of Kepler’s equation in all eccentricity regimes."
   Celestial Mechanics and Dynamical Astronomy 116, no. 1 (2013): 21-34.
```

(as explained in https://numpydoc.readthedocs.io/en/latest/format.html#sections, section 13)

Juanlu001 replied:
Also, rename `Note` to `Notes` (and adjust the `-----`).

nikita-astronaut commented:
Hello! Thanks for the review)

I added initial guess from the paper for hyperbolic region, but for non-high eccenricies. Weens like for ecc = 3200, their initial guess fails, but ours is robust, stable and works. But for smaller eccs, their initial guess works faster, so I added it. The threshold of 10 is chosen at random)

Juanlu001 commented:
Now there are some convergence failures though

nikita-astronaut commented:
With their initial guess, there are always tests not converging. The problem is with large \Delta t and ecc, cause it was not included into the paper tests. So, our initial guess, as compared to the paper, is around 1 iteration slower, but at least stable.

Juanlu001 commented:
I think this is good to go!
